{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyP8YLrpL9PDzjKV32ewxECK",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/adigenova/genome_assembly_workshop/blob/main/code/assembly_demo.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "c_Gvigr7IKu5"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Genome assembly hands-on\n",
        "\n",
        "## installing conda"
      ],
      "metadata": {
        "id": "rEpOBtSfIMMN"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget -c https://repo.continuum.io/archive/Anaconda3-5.1.0-Linux-x86_64.sh\n",
        "!chmod +x Anaconda3-5.1.0-Linux-x86_64.sh\n",
        "!bash ./Anaconda3-5.1.0-Linux-x86_64.sh -b -f -p /usr/local"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SUHoO9_2ISEj",
        "outputId": "0231a456-3389-4daf-e88d-f35e9ec227e5"
      },
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-09-15 03:11:37--  https://repo.continuum.io/archive/Anaconda3-5.1.0-Linux-x86_64.sh\n",
            "Resolving repo.continuum.io (repo.continuum.io)... 104.18.201.79, 104.18.200.79, 2606:4700::6812:c84f, ...\n",
            "Connecting to repo.continuum.io (repo.continuum.io)|104.18.201.79|:443... connected.\n",
            "HTTP request sent, awaiting response... 301 Moved Permanently\n",
            "Location: https://repo.anaconda.com/archive/Anaconda3-5.1.0-Linux-x86_64.sh [following]\n",
            "--2022-09-15 03:11:37--  https://repo.anaconda.com/archive/Anaconda3-5.1.0-Linux-x86_64.sh\n",
            "Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.130.3, 104.16.131.3, 2606:4700::6810:8203, ...\n",
            "Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.130.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 577996269 (551M) [application/x-sh]\n",
            "Saving to: ‘Anaconda3-5.1.0-Linux-x86_64.sh’\n",
            "\n",
            "Anaconda3-5.1.0-Lin 100%[===================>] 551.22M   240MB/s    in 2.3s    \n",
            "\n",
            "2022-09-15 03:11:39 (240 MB/s) - ‘Anaconda3-5.1.0-Linux-x86_64.sh’ saved [577996269/577996269]\n",
            "\n",
            "PREFIX=/usr/local\n",
            "installing: python-3.6.4-hc3d631a_1 ...\n",
            "Python 3.6.4 :: Anaconda, Inc.\n",
            "installing: ca-certificates-2017.08.26-h1d4fec5_0 ...\n",
            "installing: conda-env-2.6.0-h36134e3_1 ...\n",
            "installing: intel-openmp-2018.0.0-hc7b2577_8 ...\n",
            "installing: libgcc-ng-7.2.0-h7cc24e2_2 ...\n",
            "installing: libgfortran-ng-7.2.0-h9f7466a_2 ...\n",
            "installing: libstdcxx-ng-7.2.0-h7a57d05_2 ...\n",
            "installing: bzip2-1.0.6-h9a117a8_4 ...\n",
            "installing: expat-2.2.5-he0dffb1_0 ...\n",
            "installing: gmp-6.1.2-h6c8ec71_1 ...\n",
            "installing: graphite2-1.3.10-hf63cedd_1 ...\n",
            "installing: icu-58.2-h9c2bf20_1 ...\n",
            "installing: jbig-2.1-hdba287a_0 ...\n",
            "installing: jpeg-9b-h024ee3a_2 ...\n",
            "installing: libffi-3.2.1-hd88cf55_4 ...\n",
            "installing: libsodium-1.0.15-hf101ebd_0 ...\n",
            "installing: libtool-2.4.6-h544aabb_3 ...\n",
            "installing: libxcb-1.12-hcd93eb1_4 ...\n",
            "installing: lzo-2.10-h49e0be7_2 ...\n",
            "installing: mkl-2018.0.1-h19d6760_4 ...\n",
            "installing: ncurses-6.0-h9df7e31_2 ...\n",
            "installing: openssl-1.0.2n-hb7f436b_0 ...\n",
            "installing: patchelf-0.9-hf79760b_2 ...\n",
            "installing: pcre-8.41-hc27e229_1 ...\n",
            "installing: pixman-0.34.0-hceecf20_3 ...\n",
            "installing: tk-8.6.7-hc745277_3 ...\n",
            "installing: unixodbc-2.3.4-hc36303a_1 ...\n",
            "installing: xz-5.2.3-h55aa19d_2 ...\n",
            "installing: yaml-0.1.7-had09818_2 ...\n",
            "installing: zlib-1.2.11-ha838bed_2 ...\n",
            "installing: glib-2.53.6-h5d9569c_2 ...\n",
            "installing: hdf5-1.10.1-h9caa474_1 ...\n",
            "installing: libedit-3.1-heed3624_0 ...\n",
            "installing: libpng-1.6.34-hb9fc6fc_0 ...\n",
            "installing: libssh2-1.8.0-h9cfc8f7_4 ...\n",
            "installing: libtiff-4.0.9-h28f6b97_0 ...\n",
            "installing: libxml2-2.9.7-h26e45fe_0 ...\n",
            "installing: mpfr-3.1.5-h11a74b3_2 ...\n",
            "installing: pandoc-1.19.2.1-hea2e7c5_1 ...\n",
            "installing: readline-7.0-ha6073c6_4 ...\n",
            "installing: zeromq-4.2.2-hbedb6e5_2 ...\n",
            "installing: dbus-1.12.2-hc3f9b76_1 ...\n",
            "installing: freetype-2.8-hab7d2ae_1 ...\n",
            "installing: gstreamer-1.12.4-hb53b477_0 ...\n",
            "installing: libcurl-7.58.0-h1ad7b7a_0 ...\n",
            "installing: libxslt-1.1.32-h1312cb7_0 ...\n",
            "installing: mpc-1.0.3-hec55b23_5 ...\n",
            "installing: sqlite-3.22.0-h1bed415_0 ...\n",
            "installing: curl-7.58.0-h84994c4_0 ...\n",
            "installing: fontconfig-2.12.4-h88586e7_1 ...\n",
            "installing: gst-plugins-base-1.12.4-h33fb286_0 ...\n",
            "installing: alabaster-0.7.10-py36h306e16b_0 ...\n",
            "installing: asn1crypto-0.24.0-py36_0 ...\n",
            "installing: attrs-17.4.0-py36_0 ...\n",
            "installing: backports-1.0-py36hfa02d7e_1 ...\n",
            "installing: beautifulsoup4-4.6.0-py36h49b8c8c_1 ...\n",
            "installing: bitarray-0.8.1-py36h14c3975_1 ...\n",
            "installing: boto-2.48.0-py36h6e4cd66_1 ...\n",
            "installing: cairo-1.14.12-h77bcde2_0 ...\n",
            "installing: certifi-2018.1.18-py36_0 ...\n",
            "installing: chardet-3.0.4-py36h0f667ec_1 ...\n",
            "installing: click-6.7-py36h5253387_0 ...\n",
            "installing: cloudpickle-0.5.2-py36_1 ...\n",
            "installing: colorama-0.3.9-py36h489cec4_0 ...\n",
            "installing: contextlib2-0.5.5-py36h6c84a62_0 ...\n",
            "installing: dask-core-0.16.1-py36_0 ...\n",
            "installing: decorator-4.2.1-py36_0 ...\n",
            "installing: docutils-0.14-py36hb0f60f5_0 ...\n",
            "installing: entrypoints-0.2.3-py36h1aec115_2 ...\n",
            "installing: et_xmlfile-1.0.1-py36hd6bccc3_0 ...\n",
            "installing: fastcache-1.0.2-py36h14c3975_2 ...\n",
            "installing: filelock-2.0.13-py36h646ffb5_0 ...\n",
            "installing: glob2-0.6-py36he249c77_0 ...\n",
            "installing: gmpy2-2.0.8-py36hc8893dd_2 ...\n",
            "installing: greenlet-0.4.12-py36h2d503a6_0 ...\n",
            "installing: heapdict-1.0.0-py36_2 ...\n",
            "installing: idna-2.6-py36h82fb2a8_1 ...\n",
            "installing: imagesize-0.7.1-py36h52d8127_0 ...\n",
            "installing: ipython_genutils-0.2.0-py36hb52b0d5_0 ...\n",
            "installing: itsdangerous-0.24-py36h93cc618_1 ...\n",
            "installing: jdcal-1.3-py36h4c697fb_0 ...\n",
            "installing: lazy-object-proxy-1.3.1-py36h10fcdad_0 ...\n",
            "installing: llvmlite-0.21.0-py36ha241eea_0 ...\n",
            "installing: locket-0.2.0-py36h787c0ad_1 ...\n",
            "installing: lxml-4.1.1-py36hf71bdeb_1 ...\n",
            "installing: markupsafe-1.0-py36hd9260cd_1 ...\n",
            "installing: mccabe-0.6.1-py36h5ad9710_1 ...\n",
            "installing: mistune-0.8.3-py36_0 ...\n",
            "installing: mkl-service-1.1.2-py36h17a0993_4 ...\n",
            "installing: mpmath-1.0.0-py36hfeacd6b_2 ...\n",
            "installing: msgpack-python-0.5.1-py36h6bb024c_0 ...\n",
            "installing: multipledispatch-0.4.9-py36h41da3fb_0 ...\n",
            "installing: numpy-1.14.0-py36h3dfced4_1 ...\n",
            "installing: olefile-0.45.1-py36_0 ...\n",
            "installing: pandocfilters-1.4.2-py36ha6701b7_1 ...\n",
            "installing: parso-0.1.1-py36h35f843b_0 ...\n",
            "installing: path.py-10.5-py36h55ceabb_0 ...\n",
            "installing: pep8-1.7.1-py36_0 ...\n",
            "installing: pickleshare-0.7.4-py36h63277f8_0 ...\n",
            "installing: pkginfo-1.4.1-py36h215d178_1 ...\n",
            "installing: pluggy-0.6.0-py36hb689045_0 ...\n",
            "installing: ply-3.10-py36hed35086_0 ...\n",
            "installing: psutil-5.4.3-py36h14c3975_0 ...\n",
            "installing: ptyprocess-0.5.2-py36h69acd42_0 ...\n",
            "installing: py-1.5.2-py36h29bf505_0 ...\n",
            "installing: pycodestyle-2.3.1-py36hf609f19_0 ...\n",
            "installing: pycosat-0.6.3-py36h0a5515d_0 ...\n",
            "installing: pycparser-2.18-py36hf9f622e_1 ...\n",
            "installing: pycrypto-2.6.1-py36h14c3975_7 ...\n",
            "installing: pycurl-7.43.0.1-py36hb7f436b_0 ...\n",
            "installing: pyodbc-4.0.22-py36hf484d3e_0 ...\n",
            "installing: pyparsing-2.2.0-py36hee85983_1 ...\n",
            "installing: pysocks-1.6.7-py36hd97a5b1_1 ...\n",
            "installing: pytz-2017.3-py36h63b9c63_0 ...\n",
            "installing: pyyaml-3.12-py36hafb9ca4_1 ...\n",
            "installing: pyzmq-16.0.3-py36he2533c7_0 ...\n",
            "installing: qt-5.6.2-h974d657_12 ...\n",
            "installing: qtpy-1.3.1-py36h3691cc8_0 ...\n",
            "installing: rope-0.10.7-py36h147e2ec_0 ...\n",
            "installing: ruamel_yaml-0.15.35-py36h14c3975_1 ...\n",
            "installing: send2trash-1.4.2-py36_0 ...\n",
            "installing: simplegeneric-0.8.1-py36_2 ...\n",
            "installing: sip-4.18.1-py36h51ed4ed_2 ...\n",
            "installing: six-1.11.0-py36h372c433_1 ...\n",
            "installing: snowballstemmer-1.2.1-py36h6febd40_0 ...\n",
            "installing: sortedcontainers-1.5.9-py36_0 ...\n",
            "installing: sphinxcontrib-1.0-py36h6d0f590_1 ...\n",
            "installing: sqlalchemy-1.2.1-py36h14c3975_0 ...\n",
            "installing: tblib-1.3.2-py36h34cf8b6_0 ...\n",
            "installing: testpath-0.3.1-py36h8cadb63_0 ...\n",
            "installing: toolz-0.9.0-py36_0 ...\n",
            "installing: tornado-4.5.3-py36_0 ...\n",
            "installing: typing-3.6.2-py36h7da032a_0 ...\n",
            "installing: unicodecsv-0.14.1-py36ha668878_0 ...\n",
            "installing: wcwidth-0.1.7-py36hdf4376a_0 ...\n",
            "installing: webencodings-0.5.1-py36h800622e_1 ...\n",
            "installing: werkzeug-0.14.1-py36_0 ...\n",
            "installing: wrapt-1.10.11-py36h28b7045_0 ...\n",
            "installing: xlrd-1.1.0-py36h1db9f0c_1 ...\n",
            "installing: xlsxwriter-1.0.2-py36h3de1aca_0 ...\n",
            "installing: xlwt-1.3.0-py36h7b00a1f_0 ...\n",
            "installing: babel-2.5.3-py36_0 ...\n",
            "installing: backports.shutil_get_terminal_size-1.0.0-py36hfea85ff_2 ...\n",
            "installing: bottleneck-1.2.1-py36haac1ea0_0 ...\n",
            "installing: cffi-1.11.4-py36h9745a5d_0 ...\n",
            "installing: conda-verify-2.0.0-py36h98955d8_0 ...\n",
            "installing: cycler-0.10.0-py36h93f1223_0 ...\n",
            "installing: cytoolz-0.9.0-py36h14c3975_0 ...\n",
            "installing: h5py-2.7.1-py36h3585f63_0 ...\n",
            "installing: harfbuzz-1.7.4-hc5b324e_0 ...\n",
            "installing: html5lib-1.0.1-py36h2f9c1c0_0 ...\n",
            "installing: jedi-0.11.1-py36_0 ...\n",
            "installing: networkx-2.1-py36_0 ...\n",
            "installing: nltk-3.2.5-py36h7532b22_0 ...\n",
            "installing: numba-0.36.2-np114py36hc6662d5_0 ...\n",
            "installing: numexpr-2.6.4-py36hc4a3f9a_0 ...\n",
            "installing: openpyxl-2.4.10-py36_0 ...\n",
            "installing: packaging-16.8-py36ha668100_1 ...\n",
            "installing: partd-0.3.8-py36h36fd896_0 ...\n",
            "installing: pathlib2-2.3.0-py36h49efa8e_0 ...\n",
            "installing: pexpect-4.3.1-py36_0 ...\n",
            "installing: pillow-5.0.0-py36h3deb7b8_0 ...\n",
            "installing: pyqt-5.6.0-py36h0386399_5 ...\n",
            "installing: python-dateutil-2.6.1-py36h88d3b88_1 ...\n",
            "installing: pywavelets-0.5.2-py36he602eb0_0 ...\n",
            "installing: qtawesome-0.4.4-py36h609ed8c_0 ...\n",
            "installing: scipy-1.0.0-py36hbf646e7_0 ...\n",
            "installing: setuptools-38.4.0-py36_0 ...\n",
            "installing: singledispatch-3.4.0.3-py36h7a266c3_0 ...\n",
            "installing: sortedcollections-0.5.3-py36h3c761f9_0 ...\n",
            "installing: sphinxcontrib-websupport-1.0.1-py36hb5cb234_1 ...\n",
            "installing: sympy-1.1.1-py36hc6d1c1c_0 ...\n",
            "installing: terminado-0.8.1-py36_1 ...\n",
            "installing: traitlets-4.3.2-py36h674d592_0 ...\n",
            "installing: zict-0.1.3-py36h3a3bf81_0 ...\n",
            "installing: astroid-1.6.1-py36_0 ...\n",
            "installing: bleach-2.1.2-py36_0 ...\n",
            "installing: clyent-1.2.2-py36h7e57e65_1 ...\n",
            "installing: cryptography-2.1.4-py36hd09be54_0 ...\n",
            "installing: cython-0.27.3-py36h1860423_0 ...\n",
            "installing: datashape-0.5.4-py36h3ad6b5c_0 ...\n",
            "installing: distributed-1.20.2-py36_0 ...\n",
            "installing: get_terminal_size-1.0.0-haa9412d_0 ...\n",
            "installing: gevent-1.2.2-py36h2fe25dc_0 ...\n",
            "installing: imageio-2.2.0-py36he555465_0 ...\n",
            "installing: isort-4.2.15-py36had401c0_0 ...\n",
            "installing: jinja2-2.10-py36ha16c418_0 ...\n",
            "installing: jsonschema-2.6.0-py36h006f8b5_0 ...\n",
            "installing: jupyter_core-4.4.0-py36h7c827e3_0 ...\n",
            "installing: matplotlib-2.1.2-py36h0e671d2_0 ...\n",
            "installing: navigator-updater-0.1.0-py36h14770f7_0 ...\n",
            "installing: nose-1.3.7-py36hcdf7029_2 ...\n",
            "installing: pandas-0.22.0-py36hf484d3e_0 ...\n",
            "installing: pango-1.41.0-hd475d92_0 ...\n",
            "installing: patsy-0.5.0-py36_0 ...\n",
            "installing: pyflakes-1.6.0-py36h7bd6a15_0 ...\n",
            "installing: pygments-2.2.0-py36h0d3125c_0 ...\n",
            "installing: pytables-3.4.2-py36h3b5282a_2 ...\n",
            "installing: pytest-3.3.2-py36_0 ...\n",
            "installing: scikit-learn-0.19.1-py36h7aa7ec6_0 ...\n",
            "installing: wheel-0.30.0-py36hfd4bba0_1 ...\n",
            "installing: astropy-2.0.3-py36h14c3975_0 ...\n",
            "installing: bkcharts-0.2-py36h735825a_0 ...\n",
            "installing: bokeh-0.12.13-py36h2f9c1c0_0 ...\n",
            "installing: flask-0.12.2-py36hb24657c_0 ...\n",
            "installing: jupyter_client-5.2.2-py36_0 ...\n",
            "installing: nbformat-4.4.0-py36h31c9010_0 ...\n",
            "installing: pip-9.0.1-py36h6c6f9ce_4 ...\n",
            "installing: prompt_toolkit-1.0.15-py36h17d85b1_0 ...\n",
            "installing: pylint-1.8.2-py36_0 ...\n",
            "installing: pyopenssl-17.5.0-py36h20ba746_0 ...\n",
            "installing: statsmodels-0.8.0-py36h8533d0b_0 ...\n",
            "installing: dask-0.16.1-py36_0 ...\n",
            "installing: flask-cors-3.0.3-py36h2d857d3_0 ...\n",
            "installing: ipython-6.2.1-py36h88c514a_1 ...\n",
            "installing: nbconvert-5.3.1-py36hb41ffb7_0 ...\n",
            "installing: seaborn-0.8.1-py36hfad7ec4_0 ...\n",
            "installing: urllib3-1.22-py36hbe7ace6_0 ...\n",
            "installing: ipykernel-4.8.0-py36_0 ...\n",
            "installing: odo-0.5.1-py36h90ed295_0 ...\n",
            "installing: requests-2.18.4-py36he2e5f8d_1 ...\n",
            "installing: scikit-image-0.13.1-py36h14c3975_1 ...\n",
            "installing: anaconda-client-1.6.9-py36_0 ...\n",
            "installing: blaze-0.11.3-py36h4e06776_0 ...\n",
            "installing: jupyter_console-5.2.0-py36he59e554_1 ...\n",
            "installing: notebook-5.4.0-py36_0 ...\n",
            "installing: qtconsole-4.3.1-py36h8f73b5b_0 ...\n",
            "installing: sphinx-1.6.6-py36_0 ...\n",
            "installing: anaconda-project-0.8.2-py36h44fb852_0 ...\n",
            "installing: jupyterlab_launcher-0.10.2-py36_0 ...\n",
            "installing: numpydoc-0.7.0-py36h18f165f_0 ...\n",
            "installing: widgetsnbextension-3.1.0-py36_0 ...\n",
            "installing: anaconda-navigator-1.7.0-py36_0 ...\n",
            "installing: ipywidgets-7.1.1-py36_0 ...\n",
            "installing: jupyterlab-0.31.5-py36_0 ...\n",
            "installing: spyder-3.2.6-py36_0 ...\n",
            "installing: _ipyw_jlab_nb_ext_conf-0.1.0-py36he11e457_0 ...\n",
            "installing: jupyter-1.0.0-py36_4 ...\n",
            "installing: anaconda-5.1.0-py36_2 ...\n",
            "installing: conda-4.4.10-py36_0 ...\n",
            "installing: conda-build-3.4.1-py36_0 ...\n",
            "installation finished.\n",
            "WARNING:\n",
            "    You currently have a PYTHONPATH environment variable set. This may cause\n",
            "    unexpected behavior when running the Python interpreter in Anaconda3.\n",
            "    For best results, please verify that your PYTHONPATH only points to\n",
            "    directories of packages that are compatible with the Python interpreter\n",
            "    in Anaconda3: /usr/local\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import sys\n",
        "\n",
        "sys.path.append(\"/usr/local/lib/python3.6/site-packages/\")"
      ],
      "metadata": {
        "id": "TH3aKNToJJAf"
      },
      "execution_count": 2,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!conda config --add channels defaults\n",
        "!conda config --add channels bioconda\n",
        "!conda config --add channels conda-forge"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Z_WdNA5vJNrk",
        "outputId": "77ff3e88-154e-4e03-d9db-0e936315d1c4"
      },
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Warning: 'defaults' already in 'channels' list, moving to the top\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# we install the required software\n",
        "!conda install -y flye  velvet raven-assembler quast"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ygVbDu4YJTOx",
        "outputId": "d9a99eff-172a-43d7-9f55-87e56305d391"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "\n",
            "==> WARNING: A newer version of conda exists. <==\n",
            "  current version: 4.4.10\n",
            "  latest version: 4.14.0\n",
            "\n",
            "Please update conda by running\n",
            "\n",
            "    $ conda update -n base conda\n",
            "\n",
            "\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs: \n",
            "    - flye\n",
            "    - raven-assembler\n",
            "    - spades\n",
            "    - velvet\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    _sysroot_linux-64_curr_repodata_hack-3|      h5bd9786_13          14 KB  conda-forge\n",
            "    zlib-1.2.12                |       h166bdaf_3          92 KB  conda-forge\n",
            "    ncurses-6.3                |       h27087fc_1        1002 KB  conda-forge\n",
            "    libffi-3.3                 |       h58526e2_2          51 KB  conda-forge\n",
            "    python-3.9.0               |hffdb5ce_5_cpython        28.7 MB  conda-forge\n",
            "    readline-8.1.2             |       h0f457ee_0         291 KB  conda-forge\n",
            "    raven-assembler-1.8.1      |       h5b5514e_1         270 KB  bioconda\n",
            "    openmp-7.0.0               |       h2d50403_0         396 KB  conda-forge\n",
            "    openssl-1.1.1q             |       h166bdaf_0         2.1 MB  conda-forge\n",
            "    python_abi-3.9             |           2_cp39           4 KB  conda-forge\n",
            "    pip-22.2.2                 |     pyhd8ed1ab_0         1.5 MB  conda-forge\n",
            "    sqlite-3.39.3              |       h4ff8645_0         789 KB  conda-forge\n",
            "    tzdata-2022c               |       h191b570_0         119 KB  conda-forge\n",
            "    kernel-headers_linux-64-3.10.0|      h4a8ded7_13         953 KB  conda-forge\n",
            "    tk-8.6.12                  |       h27826a3_0         3.3 MB  conda-forge\n",
            "    spades-3.15.5              |       h95f258a_0        48.0 MB  bioconda\n",
            "    libgcc-ng-12.1.0           |      h8d9b700_16         940 KB  conda-forge\n",
            "    ld_impl_linux-64-2.36.1    |       hea4e1c9_2         667 KB  conda-forge\n",
            "    libgomp-12.1.0             |      h8d9b700_16         459 KB  conda-forge\n",
            "    _openmp_mutex-4.5          |            2_gnu          23 KB  conda-forge\n",
            "    wheel-0.37.1               |     pyhd8ed1ab_0          31 KB  conda-forge\n",
            "    xz-5.2.6                   |       h166bdaf_0         409 KB  conda-forge\n",
            "    libzlib-1.2.12             |       h166bdaf_3          65 KB  conda-forge\n",
            "    sysroot_linux-64-2.17      |      h4a8ded7_13        34.1 MB  conda-forge\n",
            "    flye-2.9.1                 |   py39h6935b12_0        35.2 MB  bioconda\n",
            "    _libgcc_mutex-0.1          |      conda_forge           3 KB  conda-forge\n",
            "    setuptools-65.3.0          |     pyhd8ed1ab_1         782 KB  conda-forge\n",
            "    llvm-meta-7.0.0            |                0           2 KB  conda-forge\n",
            "    velvet-1.2.10              |       h7132678_5         157 KB  bioconda\n",
            "    libstdcxx-ng-12.1.0        |      ha89aaad_16         4.3 MB  conda-forge\n",
            "    bzip2-1.0.8                |       h7f98852_4         484 KB  conda-forge\n",
            "    ca-certificates-2022.9.14  |       ha878542_0         152 KB  conda-forge\n",
            "    libsqlite-3.39.3           |       h753d276_0         789 KB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:       166.1 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "    _libgcc_mutex:                        0.1-conda_forge       conda-forge\n",
            "    _openmp_mutex:                        4.5-2_gnu             conda-forge\n",
            "    _sysroot_linux-64_curr_repodata_hack: 3-h5bd9786_13         conda-forge\n",
            "    flye:                                 2.9.1-py39h6935b12_0  bioconda   \n",
            "    kernel-headers_linux-64:              3.10.0-h4a8ded7_13    conda-forge\n",
            "    ld_impl_linux-64:                     2.36.1-hea4e1c9_2     conda-forge\n",
            "    libgomp:                              12.1.0-h8d9b700_16    conda-forge\n",
            "    libsqlite:                            3.39.3-h753d276_0     conda-forge\n",
            "    libzlib:                              1.2.12-h166bdaf_3     conda-forge\n",
            "    llvm-meta:                            7.0.0-0               conda-forge\n",
            "    openmp:                               7.0.0-h2d50403_0      conda-forge\n",
            "    python_abi:                           3.9-2_cp39            conda-forge\n",
            "    raven-assembler:                      1.8.1-h5b5514e_1      bioconda   \n",
            "    spades:                               3.15.5-h95f258a_0     bioconda   \n",
            "    sysroot_linux-64:                     2.17-h4a8ded7_13      conda-forge\n",
            "    tzdata:                               2022c-h191b570_0      conda-forge\n",
            "    velvet:                               1.2.10-h7132678_5     bioconda   \n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "    bzip2:                                1.0.6-h9a117a8_4                  --> 1.0.8-h7f98852_4         conda-forge\n",
            "    ca-certificates:                      2017.08.26-h1d4fec5_0             --> 2022.9.14-ha878542_0     conda-forge\n",
            "    libffi:                               3.2.1-hd88cf55_4                  --> 3.3-h58526e2_2           conda-forge\n",
            "    libgcc-ng:                            7.2.0-h7cc24e2_2                  --> 12.1.0-h8d9b700_16       conda-forge\n",
            "    libstdcxx-ng:                         7.2.0-h7a57d05_2                  --> 12.1.0-ha89aaad_16       conda-forge\n",
            "    ncurses:                              6.0-h9df7e31_2                    --> 6.3-h27087fc_1           conda-forge\n",
            "    openssl:                              1.0.2n-hb7f436b_0                 --> 1.1.1q-h166bdaf_0        conda-forge\n",
            "    pip:                                  9.0.1-py36h6c6f9ce_4              --> 22.2.2-pyhd8ed1ab_0      conda-forge\n",
            "    python:                               3.6.4-hc3d631a_1                  --> 3.9.0-hffdb5ce_5_cpython conda-forge\n",
            "    readline:                             7.0-ha6073c6_4                    --> 8.1.2-h0f457ee_0         conda-forge\n",
            "    setuptools:                           38.4.0-py36_0                     --> 65.3.0-pyhd8ed1ab_1      conda-forge\n",
            "    sqlite:                               3.22.0-h1bed415_0                 --> 3.39.3-h4ff8645_0        conda-forge\n",
            "    tk:                                   8.6.7-hc745277_3                  --> 8.6.12-h27826a3_0        conda-forge\n",
            "    wheel:                                0.30.0-py36hfd4bba0_1             --> 0.37.1-pyhd8ed1ab_0      conda-forge\n",
            "    xz:                                   5.2.3-h55aa19d_2                  --> 5.2.6-h166bdaf_0         conda-forge\n",
            "    zlib:                                 1.2.11-ha838bed_2                 --> 1.2.12-h166bdaf_3        conda-forge\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "_sysroot_linux-64_curr_repodata_hack 3: 100% 1.0/1 [00:00<00:00,  1.99it/s] \n",
            "zlib 1.2.12: 100% 1.0/1 [00:00<00:00, 13.55it/s]\n",
            "ncurses 6.3: 100% 1.0/1 [00:01<00:00,  1.36s/it]               \n",
            "libffi 3.3: 100% 1.0/1 [00:00<00:00, 18.61it/s]\n",
            "python 3.9.0: 100% 1.0/1 [00:13<00:00, 13.65s/it]               \n",
            "readline 8.1.2: 100% 1.0/1 [00:00<00:00,  4.29it/s] \n",
            "raven-assembler 1.8.1: 100% 1.0/1 [00:00<00:00,  2.35it/s]               \n",
            "openmp 7.0.0: 100% 1.0/1 [00:00<00:00,  2.20it/s]               \n",
            "openssl 1.1.1q: 100% 1.0/1 [00:00<00:00,  1.14it/s]               \n",
            "python_abi 3.9: 100% 1.0/1 [00:00<00:00, 29.39it/s]\n",
            "pip 22.2.2: 100% 1.0/1 [00:01<00:00,  1.08s/it]               \n",
            "sqlite 3.39.3: 100% 1.0/1 [00:00<00:00,  2.86it/s] \n",
            "tzdata 2022c: 100% 1.0/1 [00:00<00:00,  3.86it/s]               \n",
            "kernel-headers_linux-64 3.10.0: 100% 1.0/1 [00:01<00:00,  1.19s/it]  \n",
            "tk 8.6.12: 100% 1.0/1 [00:01<00:00,  1.77s/it]               \n",
            "spades 3.15.5: 100% 1.0/1 [00:26<00:00, 26.57s/it]                 \n",
            "libgcc-ng 12.1.0: 100% 1.0/1 [00:00<00:00,  1.51it/s]               \n",
            "ld_impl_linux-64 2.36.1: 100% 1.0/1 [00:00<00:00,  2.50it/s]               \n",
            "libgomp 12.1.0: 100% 1.0/1 [00:00<00:00,  4.79it/s] \n",
            "_openmp_mutex 4.5: 100% 1.0/1 [00:00<00:00, 15.18it/s]\n",
            "wheel 0.37.1: 100% 1.0/1 [00:00<00:00, 20.45it/s]\n",
            "xz 5.2.6: 100% 1.0/1 [00:00<00:00,  3.49it/s]              \n",
            "libzlib 1.2.12: 100% 1.0/1 [00:00<00:00, 17.47it/s]\n",
            "sysroot_linux-64 2.17: 100% 1.0/1 [00:16<00:00, 16.15s/it]              \n",
            "flye 2.9.1: 100% 1.0/1 [00:15<00:00, 15.42s/it]               \n",
            "_libgcc_mutex 0.1: 100% 1.0/1 [00:00<00:00, 22.41it/s]\n",
            "setuptools 65.3.0: 100% 1.0/1 [00:00<00:00,  1.43it/s]               \n",
            "llvm-meta 7.0.0: 100% 1.0/1 [00:00<00:00,  3.49it/s] \n",
            "velvet 1.2.10: 100% 1.0/1 [00:00<00:00,  2.32it/s]               \n",
            "libstdcxx-ng 12.1.0: 100% 1.0/1 [00:01<00:00,  1.74s/it]               \n",
            "bzip2 1.0.8: 100% 1.0/1 [00:00<00:00,  3.81it/s]               \n",
            "ca-certificates 2022.9.14: 100% 1.0/1 [00:00<00:00, 13.71it/s]\n",
            "libsqlite 3.39.3: 100% 1.0/1 [00:00<00:00,  2.65it/s]               \n",
            "Preparing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://github.com/adigenova/wengan/releases/download/v0.2/wengan-v0.2-bin-Linux.tar.gz\n",
        "!tar zxvf wengan-v0.2-bin-Linux.tar.gz\n",
        "# set WG to \n",
        "!export WG=$PWD/wengan-v0.2-bin-Linux/wengan.pl"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hx2qfLhyMNrl",
        "outputId": "6b941a89-a569-4ccd-a6c6-dda69cb87d00"
      },
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-09-15 03:35:35--  https://github.com/adigenova/wengan/releases/download/v0.2/wengan-v0.2-bin-Linux.tar.gz\n",
            "Resolving github.com (github.com)... 20.27.177.113\n",
            "Connecting to github.com (github.com)|20.27.177.113|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/173720877/6b787300-990a-11ea-885f-04d69678479f?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20220915%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220915T033451Z&X-Amz-Expires=300&X-Amz-Signature=ed51e713ee0a1e92bf40a3969097a176f1a5df3647b27d26724151ffdd28baf5&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=173720877&response-content-disposition=attachment%3B%20filename%3Dwengan-v0.2-bin-Linux.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
            "--2022-09-15 03:35:35--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/173720877/6b787300-990a-11ea-885f-04d69678479f?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20220915%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220915T033451Z&X-Amz-Expires=300&X-Amz-Signature=ed51e713ee0a1e92bf40a3969097a176f1a5df3647b27d26724151ffdd28baf5&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=173720877&response-content-disposition=attachment%3B%20filename%3Dwengan-v0.2-bin-Linux.tar.gz&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 57696888 (55M) [application/octet-stream]\n",
            "Saving to: ‘wengan-v0.2-bin-Linux.tar.gz.1’\n",
            "\n",
            "wengan-v0.2-bin-Lin 100%[===================>]  55.02M   169MB/s    in 0.3s    \n",
            "\n",
            "2022-09-15 03:35:36 (169 MB/s) - ‘wengan-v0.2-bin-Linux.tar.gz.1’ saved [57696888/57696888]\n",
            "\n",
            "wengan-v0.2-bin-Linux/\n",
            "wengan-v0.2-bin-Linux/bin/\n",
            "wengan-v0.2-bin-Linux/bin/MergePaths\n",
            "wengan-v0.2-bin-Linux/bin/abyss-junction\n",
            "wengan-v0.2-bin-Linux/bin/abyss-dida\n",
            "wengan-v0.2-bin-Linux/bin/abyss-bowtie\n",
            "wengan-v0.2-bin-Linux/bin/ParseAligns\n",
            "wengan-v0.2-bin-Linux/bin/abyss-bwamem\n",
            "wengan-v0.2-bin-Linux/bin/abyss-longseqdist\n",
            "wengan-v0.2-bin-Linux/bin/MakeLookupTable\n",
            "wengan-v0.2-bin-Linux/bin/abyss-todot\n",
            "wengan-v0.2-bin-Linux/bin/abyss-map\n",
            "wengan-v0.2-bin-Linux/bin/DAssembler\n",
            "wengan-v0.2-bin-Linux/bin/abyss-kaligner\n",
            "wengan-v0.2-bin-Linux/bin/DiscovarExp\n",
            "wengan-v0.2-bin-Linux/bin/abyss-stack-size\n",
            "wengan-v0.2-bin-Linux/bin/abyss-map-ssq\n",
            "wengan-v0.2-bin-Linux/bin/abyss-index\n",
            "wengan-v0.2-bin-Linux/bin/fastmin-sg\n",
            "wengan-v0.2-bin-Linux/bin/abyss-samtoafg\n",
            "wengan-v0.2-bin-Linux/bin/abyss-fixmate\n",
            "wengan-v0.2-bin-Linux/bin/Overlap\n",
            "wengan-v0.2-bin-Linux/bin/abyss-tabtomd\n",
            "wengan-v0.2-bin-Linux/bin/MergeContigs\n",
            "wengan-v0.2-bin-Linux/bin/abyss-bwasw\n",
            "wengan-v0.2-bin-Linux/bin/abyss-bloom\n",
            "wengan-v0.2-bin-Linux/bin/abyss-filtergraph\n",
            "wengan-v0.2-bin-Linux/bin/abyss-fatoagp\n",
            "wengan-v0.2-bin-Linux/bin/minia\n",
            "wengan-v0.2-bin-Linux/bin/abyss-bowtie2\n",
            "wengan-v0.2-bin-Linux/bin/abyss-gapfill\n",
            "wengan-v0.2-bin-Linux/bin/seqtk\n",
            "wengan-v0.2-bin-Linux/bin/abyss-bloom-dbg\n",
            "wengan-v0.2-bin-Linux/bin/abyss-align\n",
            "wengan-v0.2-bin-Linux/bin/abyss-fac\n",
            "wengan-v0.2-bin-Linux/bin/DistanceEst\n",
            "wengan-v0.2-bin-Linux/bin/abyss-mergepairs\n",
            "wengan-v0.2-bin-Linux/bin/PopBubbles\n",
            "wengan-v0.2-bin-Linux/bin/KAligner\n",
            "wengan-v0.2-bin-Linux/bin/SimpleGraph\n",
            "wengan-v0.2-bin-Linux/bin/EdgeInfo\n",
            "wengan-v0.2-bin-Linux/bin/abyss-paired-dbg\n",
            "wengan-v0.2-bin-Linux/bin/AffineAlign\n",
            "wengan-v0.2-bin-Linux/bin/PathOverlap\n",
            "wengan-v0.2-bin-Linux/bin/QueryLookupTable\n",
            "wengan-v0.2-bin-Linux/bin/NhoodInfo\n",
            "wengan-v0.2-bin-Linux/bin/FormPath\n",
            "wengan-v0.2-bin-Linux/bin/abyss-layout\n",
            "wengan-v0.2-bin-Linux/bin/CrossOut\n",
            "wengan-v0.2-bin-Linux/bin/ABYSS-P\n",
            "wengan-v0.2-bin-Linux/bin/ABYSS\n",
            "wengan-v0.2-bin-Linux/bin/DistanceEst-ssq\n",
            "wengan-v0.2-bin-Linux/bin/abyss-sealer\n",
            "wengan-v0.2-bin-Linux/bin/abyss-pe\n",
            "wengan-v0.2-bin-Linux/bin/abyss-bloom-dist.mk\n",
            "wengan-v0.2-bin-Linux/bin/konnector\n",
            "wengan-v0.2-bin-Linux/bin/abyss-gc\n",
            "wengan-v0.2-bin-Linux/bin/abyss-tofastq\n",
            "wengan-v0.2-bin-Linux/bin/CmpSeq\n",
            "wengan-v0.2-bin-Linux/bin/abyss-bwa\n",
            "wengan-v0.2-bin-Linux/bin/Consensus\n",
            "wengan-v0.2-bin-Linux/bin/intervalmiss\n",
            "wengan-v0.2-bin-Linux/bin/abyss-scaffold\n",
            "wengan-v0.2-bin-Linux/bin/PathConsensus\n",
            "wengan-v0.2-bin-Linux/bin/liger\n",
            "wengan-v0.2-bin-Linux/bin/logcounter\n",
            "wengan-v0.2-bin-Linux/bin/abyss-db-txt\n",
            "wengan-v0.2-bin-Linux/bin/AdjList\n",
            "wengan-v0.2-bin-Linux/bin/abyss-overlap\n",
            "wengan-v0.2-bin-Linux/bin/abyss-fixmate-ssq\n",
            "wengan-v0.2-bin-Linux/bin/SeqInfo\n",
            "wengan-v0.2-bin-Linux/.git/\n",
            "wengan-v0.2-bin-Linux/.git/config\n",
            "wengan-v0.2-bin-Linux/.git/description\n",
            "wengan-v0.2-bin-Linux/.git/HEAD\n",
            "wengan-v0.2-bin-Linux/.git/branches/\n",
            "wengan-v0.2-bin-Linux/.git/hooks/\n",
            "wengan-v0.2-bin-Linux/.git/hooks/pre-push.sample\n",
            "wengan-v0.2-bin-Linux/.git/hooks/update.sample\n",
            "wengan-v0.2-bin-Linux/.git/hooks/applypatch-msg.sample\n",
            "wengan-v0.2-bin-Linux/.git/hooks/pre-rebase.sample\n",
            "wengan-v0.2-bin-Linux/.git/hooks/pre-commit.sample\n",
            "wengan-v0.2-bin-Linux/.git/hooks/post-update.sample\n",
            "wengan-v0.2-bin-Linux/.git/hooks/pre-applypatch.sample\n",
            "wengan-v0.2-bin-Linux/.git/hooks/prepare-commit-msg.sample\n",
            "wengan-v0.2-bin-Linux/.git/hooks/commit-msg.sample\n",
            "wengan-v0.2-bin-Linux/.git/index\n",
            "wengan-v0.2-bin-Linux/.git/packed-refs\n",
            "wengan-v0.2-bin-Linux/.git/objects/\n",
            "wengan-v0.2-bin-Linux/.git/objects/pack/\n",
            "wengan-v0.2-bin-Linux/.git/objects/pack/pack-377872d909d3c4bf6e6985cbaaf61bc4cdb300c7.pack\n",
            "wengan-v0.2-bin-Linux/.git/objects/pack/pack-377872d909d3c4bf6e6985cbaaf61bc4cdb300c7.idx\n",
            "wengan-v0.2-bin-Linux/.git/objects/info/\n",
            "wengan-v0.2-bin-Linux/.git/logs/\n",
            "wengan-v0.2-bin-Linux/.git/logs/HEAD\n",
            "wengan-v0.2-bin-Linux/.git/logs/refs/\n",
            "wengan-v0.2-bin-Linux/.git/logs/refs/heads/\n",
            "wengan-v0.2-bin-Linux/.git/logs/refs/heads/master\n",
            "wengan-v0.2-bin-Linux/.git/logs/refs/remotes/\n",
            "wengan-v0.2-bin-Linux/.git/logs/refs/remotes/origin/\n",
            "wengan-v0.2-bin-Linux/.git/logs/refs/remotes/origin/HEAD\n",
            "wengan-v0.2-bin-Linux/.git/info/\n",
            "wengan-v0.2-bin-Linux/.git/info/exclude\n",
            "wengan-v0.2-bin-Linux/.git/refs/\n",
            "wengan-v0.2-bin-Linux/.git/refs/tags/\n",
            "wengan-v0.2-bin-Linux/.git/refs/heads/\n",
            "wengan-v0.2-bin-Linux/.git/refs/heads/master\n",
            "wengan-v0.2-bin-Linux/.git/refs/remotes/\n",
            "wengan-v0.2-bin-Linux/.git/refs/remotes/origin/\n",
            "wengan-v0.2-bin-Linux/.git/refs/remotes/origin/HEAD\n",
            "wengan-v0.2-bin-Linux/perl/\n",
            "wengan-v0.2-bin-Linux/perl/Module/\n",
            "wengan-v0.2-bin-Linux/perl/Module/Load.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Bruijn/\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Bruijn/Abyss.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Bruijn/Disco.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Bruijn/Minia.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Reads.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/SSG/\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/SSG/LG.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Scheduler/\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Scheduler/Local.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Mapper/\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Mapper/FML.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Mapper/FMS.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Pipeline/\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Pipeline/WenganM.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Pipeline/WenganA.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Pipeline/WenganD.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Chimeric/\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Chimeric/IM.pm\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Common/\n",
            "wengan-v0.2-bin-Linux/perl/Wengan/Common/GlobalConfig.pm\n",
            "wengan-v0.2-bin-Linux/wengan-diagram.svg\n",
            "wengan-v0.2-bin-Linux/share/\n",
            "wengan-v0.2-bin-Linux/share/doc/\n",
            "wengan-v0.2-bin-Linux/share/doc/abyss/\n",
            "wengan-v0.2-bin-Linux/share/doc/abyss/CITATION.bib\n",
            "wengan-v0.2-bin-Linux/share/doc/abyss/flowchart_simplified.pdf\n",
            "wengan-v0.2-bin-Linux/share/doc/abyss/ChangeLog\n",
            "wengan-v0.2-bin-Linux/share/doc/abyss/CITATION.md\n",
            "wengan-v0.2-bin-Linux/share/doc/abyss/COPYRIGHT\n",
            "wengan-v0.2-bin-Linux/share/doc/abyss/README.md\n",
            "wengan-v0.2-bin-Linux/share/doc/abyss/flowchart.pdf\n",
            "wengan-v0.2-bin-Linux/share/doc/abyss/LICENSE\n",
            "wengan-v0.2-bin-Linux/share/doc/discovarexp/\n",
            "wengan-v0.2-bin-Linux/share/doc/discovarexp/README\n",
            "wengan-v0.2-bin-Linux/share/doc/discovarexp/DiscovarExp\n",
            "wengan-v0.2-bin-Linux/share/doc/discovarexp/NhoodInfo\n",
            "wengan-v0.2-bin-Linux/share/doc/discovarexp/CrossOut\n",
            "wengan-v0.2-bin-Linux/share/doc/discovarexp/INSTALL\n",
            "wengan-v0.2-bin-Linux/share/doc/discovarexp/LICENSE\n",
            "wengan-v0.2-bin-Linux/share/man/\n",
            "wengan-v0.2-bin-Linux/share/man/man1/\n",
            "wengan-v0.2-bin-Linux/share/man/man1/abyss-pe.1\n",
            "wengan-v0.2-bin-Linux/share/man/man1/abyss-tofastq.1\n",
            "wengan-v0.2-bin-Linux/share/man/man1/ABYSS.1\n",
            "wengan-v0.2-bin-Linux/README.md\n",
            "wengan-v0.2-bin-Linux/.gitmodules\n",
            "wengan-v0.2-bin-Linux/LICENSE\n",
            "wengan-v0.2-bin-Linux/aux_scripts/\n",
            "wengan-v0.2-bin-Linux/aux_scripts/run-minia.pl\n",
            "wengan-v0.2-bin-Linux/wengan.pl\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! perl /content/wengan-v0.2-bin-Linux/wengan.pl \n",
        "! flye\n",
        "! raven\n",
        "! velveth"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "26T2zXkqN4Cp",
        "outputId": "0d5e9a9e-5839-4181-8b1e-8f7fad1d35a9"
      },
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "  Usage example :\n",
            "    # Assembling Oxford nanopore and illumina reads with WenganM\n",
            "    wengan.pl -x ontraw -a M -s lib1.fwd.fastq.gz,lib1.rev.fastq.gz -l ont.fastq.gz -p asm1 -t 20 -g 3000\n",
            "\n",
            "  Wengan options:\n",
            "\n",
            "   Mandatory options***:\n",
            "      -x preset [ontlon,ontraw,pacraw,ccsont,ccspac]\n",
            "      -a Mode [M,A,D]\n",
            "      -s Short-reads [fwd1.fastq.gz,rev1.fastq.gz..]\n",
            "      -l Long-reads.fq.gz\n",
            "      -b Hifi-reads [hifi.fq.gz,hifi2.fq.gz..]\n",
            "      -g 3000 [genome size in Mb]\n",
            "      -p prefix\n",
            "\n",
            "   General Options :\n",
            "      -h [detail information]\n",
            "      -t cores [1]\n",
            "      -c <pre-assembled short-read contigs>\n",
            "      -i <insert size lists for raw long-reads>\n",
            "      -I <insert size list for CCS reads>\n",
            "      -n <show pipeline comands>;\n",
            "\n",
            "   Advanced Options (Change the presets):\n",
            "      FastMin-SG options:\n",
            "        -k k-mer size [15-28]\n",
            "        -w minimizer window [5-15]\n",
            "        -q minimum mapping quality [20-60]\n",
            "        -m moving window [150]\n",
            "      IntervalMiss options:\n",
            "        -d Minimum base coverage [def:7]\n",
            "        -f Factor to compute std using avg*fst [def:0.1]\n",
            "      Liger options:\n",
            "        -M Minimum contig length in TR [def:2000]\n",
            "        -R Repeat copy number factor [def:1.5]\n",
            "        -L Length of long mate-edges [def:100000]\n",
            "        -N Number of long-read needed to keep a potencial erroneus mate-edge [def:5]\n",
            "        -P Minimum length of reduced paths to convert them to physical fragments [def:20kb]\n",
            "        -Q Minimum contig length in matching [def:2000]\n",
            "        -U Repeat copy number factor backbone [def:1.5]\n",
            "        -T Number of iterations in TR per edge [def:1M]\n",
            "        HiFi options:\n",
            "        -D Ploidy for Hifi hybrid assembly [haploid=1,diploid=2]\n",
            "\n",
            " /content/wengan-v0.2-bin-Linux/wengan.pl -h for detailed usage information.\n",
            "   \n",
            "usage: flye (--pacbio-raw | --pacbio-corr | --pacbio-hifi | --nano-raw |\n",
            "\t     --nano-corr | --nano-hq ) file1 [file_2 ...]\n",
            "\t     --out-dir PATH\n",
            "\n",
            "\t     [--genome-size SIZE] [--threads int] [--iterations int]\n",
            "\t     [--meta] [--polish-target] [--min-overlap SIZE]\n",
            "\t     [--keep-haplotypes] [--debug] [--version] [--help] \n",
            "\t     [--scaffold] [--resume] [--resume-from] [--stop-after] \n",
            "\t     [--read-error float] [--extra-params]\n",
            "flye: error: the following arguments are required: -o/--out-dir\n",
            "/bin/bash: spades: command not found\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Getting the data\n"
      ],
      "metadata": {
        "id": "ntI1JiJYPLZk"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!git clone https://github.com/adigenova/wengan_demo.git"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "WnGvNrI4PPsr",
        "outputId": "83ce388f-491c-4166-9f92-1745707a16cc"
      },
      "execution_count": 21,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'wengan_demo'...\n",
            "remote: Enumerating objects: 35, done.\u001b[K\n",
            "remote: Counting objects: 100% (24/24), done.\u001b[K\n",
            "remote: Compressing objects: 100% (17/17), done.\u001b[K\n",
            "remote: Total 35 (delta 7), reused 24 (delta 7), pack-reused 11\u001b[K\n",
            "Unpacking objects: 100% (35/35), done.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Running long-reads assemblers"
      ],
      "metadata": {
        "id": "fo8HufD0PYs9"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Flye"
      ],
      "metadata": {
        "id": "QeYmMg9iPfRn"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! flye --help"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0SeON3YAPdZw",
        "outputId": "3b7fbeef-fc9a-4da3-b750-242dbf796aa3"
      },
      "execution_count": 23,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "usage: flye (--pacbio-raw | --pacbio-corr | --pacbio-hifi | --nano-raw |\n",
            "\t     --nano-corr | --nano-hq ) file1 [file_2 ...]\n",
            "\t     --out-dir PATH\n",
            "\n",
            "\t     [--genome-size SIZE] [--threads int] [--iterations int]\n",
            "\t     [--meta] [--polish-target] [--min-overlap SIZE]\n",
            "\t     [--keep-haplotypes] [--debug] [--version] [--help] \n",
            "\t     [--scaffold] [--resume] [--resume-from] [--stop-after] \n",
            "\t     [--read-error float] [--extra-params]\n",
            "\n",
            "Assembly of long reads with repeat graphs\n",
            "\n",
            "optional arguments:\n",
            "  -h, --help\n",
            "    show this\n",
            "    help\n",
            "    message and\n",
            "    exit\n",
            "  --pacbio-raw path [path ...]\n",
            "    PacBio\n",
            "    regular CLR\n",
            "    reads (<20%\n",
            "    error)\n",
            "  --pacbio-corr path [path ...]\n",
            "    PacBio\n",
            "    reads that\n",
            "    were\n",
            "    corrected\n",
            "    with other\n",
            "    methods\n",
            "    (<3% error)\n",
            "  --pacbio-hifi path [path ...]\n",
            "    PacBio HiFi\n",
            "    reads (<1%\n",
            "    error)\n",
            "  --nano-raw path [path ...]\n",
            "    ONT regular\n",
            "    reads, pre-\n",
            "    Guppy5\n",
            "    (<20%\n",
            "    error)\n",
            "  --nano-corr path [path ...]\n",
            "    ONT reads\n",
            "    that were\n",
            "    corrected\n",
            "    with other\n",
            "    methods\n",
            "    (<3% error)\n",
            "  --nano-hq path [path ...]\n",
            "    ONT high-\n",
            "    quality\n",
            "    reads:\n",
            "    Guppy5+ SUP\n",
            "    or Q20 (<5%\n",
            "    error)\n",
            "  --subassemblies path [path ...]\n",
            "    [deprecated\n",
            "    ] high-\n",
            "    quality\n",
            "    contigs\n",
            "    input\n",
            "  -g size, --genome-size size\n",
            "    estimated\n",
            "    genome size\n",
            "    (for\n",
            "    example, 5m\n",
            "    or 2.6g)\n",
            "  -o path, --out-dir path\n",
            "    Output\n",
            "    directory\n",
            "  -t int, --threads int\n",
            "    number of\n",
            "    parallel\n",
            "    threads [1]\n",
            "  -i int, --iterations int\n",
            "    number of\n",
            "    polishing\n",
            "    iterations\n",
            "    [1]\n",
            "  -m int, --min-overlap int\n",
            "    minimum\n",
            "    overlap\n",
            "    between\n",
            "    reads\n",
            "    [auto]\n",
            "  --asm-coverage int\n",
            "    reduced\n",
            "    coverage\n",
            "    for initial\n",
            "    disjointig\n",
            "    assembly\n",
            "    [not set]\n",
            "  --hifi-error float\n",
            "    [deprecated\n",
            "    ] same as\n",
            "    --read-\n",
            "    error\n",
            "  --read-error float\n",
            "    adjust\n",
            "    parameters\n",
            "    for given\n",
            "    read error\n",
            "    rate (as\n",
            "    fraction\n",
            "    e.g. 0.03)\n",
            "  --extra-params extra_params\n",
            "    extra confi\n",
            "    guration\n",
            "    parameters\n",
            "    list\n",
            "    (comma-\n",
            "    separated)\n",
            "  --plasmids\n",
            "    unused\n",
            "    (retained\n",
            "    for\n",
            "    backward co\n",
            "    mpatibility\n",
            "    )\n",
            "  --meta\n",
            "    metagenome\n",
            "    / uneven\n",
            "    coverage\n",
            "    mode\n",
            "  --keep-haplotypes\n",
            "    do not\n",
            "    collapse\n",
            "    alternative\n",
            "    haplotypes\n",
            "  --no-alt-contigs\n",
            "    do not\n",
            "    output\n",
            "    contigs rep\n",
            "    resenting\n",
            "    alternative\n",
            "    haplotypes\n",
            "  --scaffold\n",
            "    enable\n",
            "    scaffolding\n",
            "    using graph\n",
            "    [disabled\n",
            "    by default]\n",
            "  --trestle\n",
            "    [deprecated\n",
            "    ] enable\n",
            "    Trestle\n",
            "    [disabled\n",
            "    by default]\n",
            "  --polish-target path\n",
            "    run\n",
            "    polisher on\n",
            "    the target\n",
            "    sequence\n",
            "  --resume\n",
            "    resume from\n",
            "    the last\n",
            "    completed\n",
            "    stage\n",
            "  --resume-from stage_name\n",
            "    resume from\n",
            "    a custom\n",
            "    stage\n",
            "  --stop-after stage_name\n",
            "    stop after\n",
            "    the\n",
            "    specified\n",
            "    stage\n",
            "    completed\n",
            "  --debug\n",
            "    enable\n",
            "    debug\n",
            "    output\n",
            "  -v, --version\n",
            "    show\n",
            "    program's\n",
            "    version\n",
            "    number and\n",
            "    exit\n",
            "\n",
            "Input reads can be in FASTA or FASTQ format, uncompressed\n",
            "or compressed with gz. Currently, PacBio (CLR, HiFi, corrected)\n",
            "and ONT reads (regular, HQ, corrected) are supported. Expected error rates are\n",
            "<15% for PB CLR/regular ONT; <5% for ONT HQ, <3% for corrected, and <1% for HiFi. Note that Flye\n",
            "was primarily developed to run on uncorrected reads. You may specify multiple\n",
            "files with reads (separated by spaces). Mixing different read\n",
            "types is not yet supported. The --meta option enables the mode\n",
            "for metagenome/uneven coverage assembly.\n",
            "\n",
            "To reduce memory consumption for large genome assemblies,\n",
            "you can use a subset of the longest reads for initial disjointig\n",
            "assembly by specifying --asm-coverage and --genome-size options. Typically,\n",
            "40x coverage is enough to produce good disjointigs.\n",
            "\n",
            "You can run Flye polisher as a standalone tool using\n",
            "--polish-target option.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# take 30 minutes\n",
        "!  flye --nano-raw /content/wengan_demo/ecoli/reads/EC.ONT.30X.fa.gz  --threads 2 --out-dir flye-ont"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QtUHOmcoPsPO",
        "outputId": "81741d21-3d0f-4c9d-ad8d-ff015895dbc6"
      },
      "execution_count": 24,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[2022-09-15 03:45:18] INFO: Starting Flye 2.9.1-b1780\n",
            "[2022-09-15 03:45:18] INFO: >>>STAGE: configure\n",
            "[2022-09-15 03:45:18] INFO: Configuring run\n",
            "[2022-09-15 03:45:20] INFO: Total read length: 149979401\n",
            "[2022-09-15 03:45:20] INFO: Reads N50/N90: 46606 / 39219\n",
            "[2022-09-15 03:45:20] INFO: Minimum overlap set to 10000\n",
            "[2022-09-15 03:45:20] INFO: >>>STAGE: assembly\n",
            "[2022-09-15 03:45:20] INFO: Assembling disjointigs\n",
            "[2022-09-15 03:45:20] INFO: Reading sequences\n",
            "tcmalloc: large alloc 8589934592 bytes == 0x564bf241a000 @  0x7f7d339da887 0x564beb2a5585 0x564beb271e90 0x564beb24d872 0x7f7d32fffc87 0x564beb24da19\n",
            "[2022-09-15 03:45:29] INFO: Counting k-mers:\n",
            "0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100% \n",
            "[2022-09-15 03:47:09] INFO: Filling index table (1/2)\n",
            "0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100% \n",
            "[2022-09-15 03:48:24] INFO: Filling index table (2/2)\n",
            "0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100% \n",
            "[2022-09-15 03:50:28] INFO: Extending reads\n",
            "[2022-09-15 03:51:24] INFO: Overlap-based coverage: 24\n",
            "[2022-09-15 03:51:24] INFO: Median overlap divergence: 0.173796\n",
            "0% 90% 100% \n",
            "[2022-09-15 03:52:00] INFO: Assembled 1 disjointigs\n",
            "[2022-09-15 03:52:01] INFO: Generating sequence\n",
            "0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100% \n",
            "[2022-09-15 03:52:24] INFO: Filtering contained disjointigs\n",
            "0% 100% \n",
            "[2022-09-15 03:52:28] INFO: Contained seqs: 0\n",
            "[2022-09-15 03:52:28] INFO: >>>STAGE: consensus\n",
            "[2022-09-15 03:52:28] INFO: Running Minimap2\n",
            "[2022-09-15 03:54:21] INFO: Computing consensus\n",
            "[2022-09-15 03:57:14] INFO: Alignment error rate: 0.242177\n",
            "[2022-09-15 03:57:14] INFO: >>>STAGE: repeat\n",
            "[2022-09-15 03:57:14] INFO: Building and resolving repeat graph\n",
            "[2022-09-15 03:57:14] INFO: Parsing disjointigs\n",
            "[2022-09-15 03:57:14] INFO: Building repeat graph\n",
            "0% 100% \n",
            "[2022-09-15 03:57:26] INFO: Median overlap divergence: 0.0482795\n",
            "[2022-09-15 03:57:26] INFO: Parsing reads\n",
            "[2022-09-15 03:57:28] INFO: Aligning reads to the graph\n",
            "0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100% \n",
            "[2022-09-15 03:58:39] INFO: Aligned read sequence: 147711009 / 149979401 (0.984875)\n",
            "[2022-09-15 03:58:39] INFO: Median overlap divergence: 0.105581\n",
            "[2022-09-15 03:58:39] INFO: Mean edge coverage: 33\n",
            "[2022-09-15 03:58:39] INFO: Simplifying the graph\n",
            "[2022-09-15 03:58:39] INFO: >>>STAGE: contigger\n",
            "[2022-09-15 03:58:39] INFO: Generating contigs\n",
            "[2022-09-15 03:58:39] INFO: Reading sequences\n",
            "[2022-09-15 03:58:41] INFO: Generated 1 contigs\n",
            "[2022-09-15 03:58:41] INFO: Added 0 scaffold connections\n",
            "[2022-09-15 03:58:41] INFO: >>>STAGE: polishing\n",
            "[2022-09-15 03:58:41] INFO: Polishing genome (1/1)\n",
            "[2022-09-15 03:58:41] INFO: Running minimap2\n",
            "[2022-09-15 03:59:52] INFO: Separating alignment into bubbles\n",
            "[2022-09-15 04:03:12] INFO: Alignment error rate: 0.175104\n",
            "[2022-09-15 04:03:12] INFO: Correcting bubbles\n",
            "0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100% \n",
            "[2022-09-15 04:16:46] INFO: >>>STAGE: finalize\n",
            "[2022-09-15 04:16:46] INFO: Assembly statistics:\n",
            "\n",
            "\tTotal length:\t4594810\n",
            "\tFragments:\t1\n",
            "\tFragments N50:\t4594810\n",
            "\tLargest frg:\t4594810\n",
            "\tScaffolds:\t0\n",
            "\tMean coverage:\t33\n",
            "\n",
            "[2022-09-15 04:16:46] INFO: Final assembly: /content/flye-ont/assembly.fasta\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Raven"
      ],
      "metadata": {
        "id": "jXQ8bV3qVvSP"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! raven "
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "D2OQ6f3OWvmM",
        "outputId": "d825c4ee-7f39-4801-838e-615c77860c71"
      },
      "execution_count": 25,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "usage: raven [options ...] <sequences> [<sequences> ...]\n",
            "\n",
            "  # default output is to stdout in FASTA format\n",
            "  <sequences>\n",
            "    input file in FASTA/FASTQ format (can be compressed with gzip)\n",
            "\n",
            "  options:\n",
            "    -k, --kmer-len <int>\n",
            "      default: 15\n",
            "      length of minimizers used to find overlaps\n",
            "    -w, --window-len <int>\n",
            "      default: 5\n",
            "      length of sliding window from which minimizers are sampled\n",
            "    -f, --frequency <double>\n",
            "      default: 0.001\n",
            "      threshold for ignoring most frequent minimizers\n",
            "    -p, --polishing-rounds <int>\n",
            "      default: 2\n",
            "      number of times racon is invoked\n",
            "    -m, --match <int>\n",
            "      default: 3\n",
            "      score for matching bases\n",
            "    -n, --mismatch <int>\n",
            "      default: -5\n",
            "      score for mismatching bases\n",
            "    -g, --gap <int>\n",
            "      default: -4\n",
            "      gap penalty (must be negative)\n",
            "    --graphical-fragment-assembly <string>\n",
            "      prints the assembly graph in GFA format\n",
            "    --resume\n",
            "      resume previous run from last checkpoint\n",
            "    --disable-checkpoints\n",
            "      disable checkpoint file creation\n",
            "    -t, --threads <int>\n",
            "      default: 1\n",
            "      number of threads\n",
            "    --version\n",
            "      prints the version number\n",
            "    -h, --help\n",
            "      prints the usage\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# 20 minutos\n",
        "! raven -t 2  /content/wengan_demo/ecoli/reads/EC.ONT.30X.fa.gz  > racon_ont.fa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iO9PXAAhX05j",
        "outputId": "185c7453-8d23-465c-cf85-9e3b85d90ba8"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[raven::] loaded 3116 sequences 1.474742s\n",
            "[raven::Graph::Construct] minimized 0 - 3116 / 3116 6.219746s\n",
            "[raven::Graph::Construct] mapped sequences 11.187644s\n",
            "[raven::Graph::Construct] annotated piles 0.133910s\n",
            "[raven::Graph::Construct] removed contained sequences 0.008196s\n",
            "[raven::Graph::Construct] removed chimeric sequences 0.001460s\n",
            "[raven::Graph::Construct] reached checkpoint 0.025432s\n",
            "[raven::Graph::Construct] minimized 0 - 443 / 443 2.433310s\n",
            "[raven::Graph::Construct] mapped valid sequences 5.170379s\n",
            "[raven::Graph::Construct] updated overlaps 0.000125s\n",
            "[raven::Graph::Construct] removed false overlaps 0.036969s\n",
            "[raven::Graph::Construct] stored 850 nodes 0.163060s\n",
            "[raven::Graph::Construct] stored 4980 edges 0.000000s\n",
            "[raven::Graph::Construct] reached checkpoint 0.122946s\n",
            "[raven::Graph::Construct] 25.518154s\n",
            "[raven::Graph::Assemble] removed transitive edges 0.002177s\n",
            "[raven::Graph::Assemble] reached checkpoint 0.131314s\n",
            "[raven::Graph::Assemble] removed tips and bubbles 0.000094s\n",
            "[raven::Graph::Assemble] reached checkpoint 0.122353s\n",
            "[raven::Graph::Assemble] removed long edges 0.077720s\n",
            "[raven::Graph::Assemble] reached checkpoint 0.092711s\n",
            "[raven::Graph::Assemble] 0.427191s\n",
            "[racon::Polisher::Polish] minimized 0 - 1 / 1 0.475131s\n",
            "[racon::Polisher::Polish] mapped sequences 17.362693s\n",
            "[racon::Polisher::Polish] found 3116 overlaps 17.902474s\n",
            "[racon::Polisher::Polish] reverse complemented sequences 0.000088s\n",
            "[racon::Polisher::Polish] aligned 3116 / 3116 overlaps [================] 422.884815s\n",
            "[racon::Polisher::Polish] prepared 8807 window placeholders 0.049460s\n",
            "[racon::Polisher::Polish] called consensus for 8807 / 8807 windows [================] 206.025727s\n",
            "[racon::Polisher::Polish] 646.894847s\n",
            "[raven::Graph::Polish] reached checkpoint 0.008451s\n",
            "[racon::Polisher::Polish] minimized 0 - 1 / 1 0.479417s\n",
            "[racon::Polisher::Polish] mapped sequences 18.645536s\n",
            "[racon::Polisher::Polish] found 3116 overlaps 19.193460s\n",
            "[racon::Polisher::Polish] reverse complemented sequences 0.000056s\n",
            "[racon::Polisher::Polish] aligned 3116 / 3116 overlaps [================] 303.612402s\n",
            "[racon::Polisher::Polish] prepared 9178 window placeholders 0.058798s\n",
            "[racon::Polisher::Polish] called consensus for 9178 / 9178 windows [================] 204.099507s\n",
            "[racon::Polisher::Polish] 526.988512s\n",
            "[raven::Graph::Polish] reached checkpoint 0.005175s\n",
            "[raven::] 1201.825915s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Short reads assemblers"
      ],
      "metadata": {
        "id": "zX_kX1cIbgji"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### velvet"
      ],
      "metadata": {
        "id": "6MvcMEEubjTF"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! velveth Assem 31 -shortPaired -fastq -separate /content/wengan_demo/ecoli/reads/EC.50X.R1.fastq.gz /content/wengan_demo/ecoli/reads/EC.50X.R2.fastq.gz "
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uLOthowpblSx",
        "outputId": "03ea8efd-51dd-4e11-b00f-450c6c908402"
      },
      "execution_count": 27,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[0.000000] Reading FastQ file /content/wengan_demo/ecoli/reads/EC.50X.R1.fastq.gz;\n",
            "[0.000343] Reading FastQ file /content/wengan_demo/ecoli/reads/EC.50X.R2.fastq.gz;\n",
            "[9.475259] 831196 sequences found in total in the paired sequence files\n",
            "[9.475287] Done\n",
            "[9.772792] Reading read set file Assem/Sequences;\n",
            "[10.057243] 831196 sequences found\n",
            "[11.486519] Done\n",
            "[11.486551] 831196 sequences in total.\n",
            "[11.486654] Writing into roadmap file Assem/Roadmaps...\n",
            "[19.176741] Inputting sequences...\n",
            "[19.179271] Inputting sequence 0 / 831196\n",
            "[119.911840]  === Sequences loaded in 100.735117 s\n",
            "[119.911932] Done inputting sequences\n",
            "[119.911941] Destroying splay table\n",
            "[129.489778] Splay table destroyed\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! velvetg Assem -cov_cutoff auto -scaffolding yes -ins_length 400 "
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "d6J1VuvVc7zo",
        "outputId": "a3e56270-228a-40ee-f235-4d6d6077fb6a"
      },
      "execution_count": 28,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[0.000000] Reading roadmap file Assem/Roadmaps\n",
            "[2.096365] 831196 roadmaps read\n",
            "[2.103263] Creating insertion markers\n",
            "[2.520192] Ordering insertion markers\n",
            "[3.455007] Counting preNodes\n",
            "[3.806986] 2363221 preNodes counted, creating them now\n",
            "[11.805138] Adjusting marker info...\n",
            "[12.097690] Connecting preNodes\n",
            "[17.544237] Cleaning up memory\n",
            "[17.550476] Done creating preGraph\n",
            "[17.550505] Concatenation...\n",
            "[18.593102] Renumbering preNodes\n",
            "[18.593150] Initial preNode count 2363221\n",
            "[18.824544] Destroyed 1134556 preNodes\n",
            "[18.824589] Concatenation over!\n",
            "[18.824594] Clipping short tips off preGraph\n",
            "[19.048076] Concatenation...\n",
            "[19.397225] Renumbering preNodes\n",
            "[19.397268] Initial preNode count 1228665\n",
            "[19.600866] Destroyed 57592 preNodes\n",
            "[19.600903] Concatenation over!\n",
            "[19.600907] 31215 tips cut off\n",
            "[19.600910] 1171073 nodes left\n",
            "[19.601005] Writing into pregraph file Assem/PreGraph...\n",
            "[22.812128] Reading read set file Assem/Sequences;\n",
            "[23.080116] 831196 sequences found\n",
            "[24.251611] Done\n",
            "[28.452396] Reading pre-graph file Assem/PreGraph\n",
            "[28.454707] Graph has 1171073 nodes and 831196 sequences\n",
            "[30.327098] Scanning pre-graph file Assem/PreGraph for k-mers\n",
            "[31.571121] 20171752 kmers found\n",
            "tcmalloc: large alloc 1210310656 bytes == 0x55dc90022000 @  0x7f973d7fe001 0x55dc753d7794 0x55dc753d8193 0x55dc753cec9a 0x55dc753d1094 0x55dc753b132f 0x7f973ce3cc87 0x55dc753b1b41\n",
            "[35.623647] Sorting kmer occurence table ... \n",
            "[57.554470] Sorting done.\n",
            "[57.554511] Computing acceleration table... \n",
            "[57.837012] Computing offsets... \n",
            "[58.070268] Ghost Threading through reads 0 / 831196\n",
            "[125.936911]  === Ghost-Threaded in 67.869826 s\n",
            "[125.937131] Threading through reads 0 / 831196\n",
            "[198.027803]  === Threaded in 72.090854 s\n",
            "[198.080806] Correcting graph with cutoff 0.200000\n",
            "[198.142347] Determining eligible starting points\n",
            "[200.637320] Done listing starting nodes\n",
            "[200.637363] Initializing todo lists\n",
            "[201.086202] Done with initilization\n",
            "[201.086245] Activating arc lookup table\n",
            "[202.024105] Done activating arc lookup table\n",
            "[202.145922] 10000 / 1171073 nodes visited\n",
            "[202.315764] 20000 / 1171073 nodes visited\n",
            "[202.500883] 30000 / 1171073 nodes visited\n",
            "[202.702867] 40000 / 1171073 nodes visited\n",
            "[202.904134] 50000 / 1171073 nodes visited\n",
            "[203.112706] 60000 / 1171073 nodes visited\n",
            "[203.312897] 70000 / 1171073 nodes visited\n",
            "[203.529879] 80000 / 1171073 nodes visited\n",
            "[203.741763] 90000 / 1171073 nodes visited\n",
            "[203.961327] 100000 / 1171073 nodes visited\n",
            "[204.174002] 110000 / 1171073 nodes visited\n",
            "[204.403810] 120000 / 1171073 nodes visited\n",
            "[204.616963] 130000 / 1171073 nodes visited\n",
            "[204.816585] 140000 / 1171073 nodes visited\n",
            "[205.029247] 150000 / 1171073 nodes visited\n",
            "[205.233232] 160000 / 1171073 nodes visited\n",
            "[205.440930] 170000 / 1171073 nodes visited\n",
            "[205.643711] 180000 / 1171073 nodes visited\n",
            "[205.842289] 190000 / 1171073 nodes visited\n",
            "[206.054224] 200000 / 1171073 nodes visited\n",
            "[206.250491] 210000 / 1171073 nodes visited\n",
            "[206.451210] 220000 / 1171073 nodes visited\n",
            "[206.650591] 230000 / 1171073 nodes visited\n",
            "[206.852753] 240000 / 1171073 nodes visited\n",
            "[207.068866] 250000 / 1171073 nodes visited\n",
            "[207.270145] 260000 / 1171073 nodes visited\n",
            "[207.467096] 270000 / 1171073 nodes visited\n",
            "[207.665092] 280000 / 1171073 nodes visited\n",
            "[207.874001] 290000 / 1171073 nodes visited\n",
            "[208.083221] 300000 / 1171073 nodes visited\n",
            "[208.292085] 310000 / 1171073 nodes visited\n",
            "[208.491507] 320000 / 1171073 nodes visited\n",
            "[208.696919] 330000 / 1171073 nodes visited\n",
            "[208.904895] 340000 / 1171073 nodes visited\n",
            "[209.120683] 350000 / 1171073 nodes visited\n",
            "[209.333657] 360000 / 1171073 nodes visited\n",
            "[209.532142] 370000 / 1171073 nodes visited\n",
            "[209.726251] 380000 / 1171073 nodes visited\n",
            "[209.950179] 390000 / 1171073 nodes visited\n",
            "[210.157160] 400000 / 1171073 nodes visited\n",
            "[210.370871] 410000 / 1171073 nodes visited\n",
            "[210.576687] 420000 / 1171073 nodes visited\n",
            "[210.781676] 430000 / 1171073 nodes visited\n",
            "[210.976963] 440000 / 1171073 nodes visited\n",
            "[211.192428] 450000 / 1171073 nodes visited\n",
            "[211.403660] 460000 / 1171073 nodes visited\n",
            "[211.608574] 470000 / 1171073 nodes visited\n",
            "[211.806865] 480000 / 1171073 nodes visited\n",
            "[212.012103] 490000 / 1171073 nodes visited\n",
            "[212.214463] 500000 / 1171073 nodes visited\n",
            "[212.419369] 510000 / 1171073 nodes visited\n",
            "[212.618151] 520000 / 1171073 nodes visited\n",
            "[212.820278] 530000 / 1171073 nodes visited\n",
            "[213.015811] 540000 / 1171073 nodes visited\n",
            "[213.228992] 550000 / 1171073 nodes visited\n",
            "[213.428129] 560000 / 1171073 nodes visited\n",
            "[213.621019] 570000 / 1171073 nodes visited\n",
            "[213.814772] 580000 / 1171073 nodes visited\n",
            "[214.010904] 590000 / 1171073 nodes visited\n",
            "[214.217552] 600000 / 1171073 nodes visited\n",
            "[214.413138] 610000 / 1171073 nodes visited\n",
            "[214.598962] 620000 / 1171073 nodes visited\n",
            "[214.801441] 630000 / 1171073 nodes visited\n",
            "[214.993937] 640000 / 1171073 nodes visited\n",
            "[215.202073] 650000 / 1171073 nodes visited\n",
            "[215.401305] 660000 / 1171073 nodes visited\n",
            "[215.596280] 670000 / 1171073 nodes visited\n",
            "[215.786731] 680000 / 1171073 nodes visited\n",
            "[215.980951] 690000 / 1171073 nodes visited\n",
            "[216.164104] 700000 / 1171073 nodes visited\n",
            "[216.365926] 710000 / 1171073 nodes visited\n",
            "[216.560523] 720000 / 1171073 nodes visited\n",
            "[216.751319] 730000 / 1171073 nodes visited\n",
            "[216.946320] 740000 / 1171073 nodes visited\n",
            "[217.145727] 750000 / 1171073 nodes visited\n",
            "[217.336227] 760000 / 1171073 nodes visited\n",
            "[217.529311] 770000 / 1171073 nodes visited\n",
            "[217.710293] 780000 / 1171073 nodes visited\n",
            "[217.899385] 790000 / 1171073 nodes visited\n",
            "[218.079888] 800000 / 1171073 nodes visited\n",
            "[218.273777] 810000 / 1171073 nodes visited\n",
            "[218.454942] 820000 / 1171073 nodes visited\n",
            "[218.643254] 830000 / 1171073 nodes visited\n",
            "[218.823130] 840000 / 1171073 nodes visited\n",
            "[219.006630] 850000 / 1171073 nodes visited\n",
            "[219.180733] 860000 / 1171073 nodes visited\n",
            "[219.369373] 870000 / 1171073 nodes visited\n",
            "[219.548932] 880000 / 1171073 nodes visited\n",
            "[219.726918] 890000 / 1171073 nodes visited\n",
            "[219.897451] 900000 / 1171073 nodes visited\n",
            "[220.067676] 910000 / 1171073 nodes visited\n",
            "[220.224572] 920000 / 1171073 nodes visited\n",
            "[220.478551] 930000 / 1171073 nodes visited\n",
            "[220.663937] 940000 / 1171073 nodes visited\n",
            "[220.880885] 950000 / 1171073 nodes visited\n",
            "[221.047984] 960000 / 1171073 nodes visited\n",
            "[221.256329] 970000 / 1171073 nodes visited\n",
            "[221.514146] 980000 / 1171073 nodes visited\n",
            "[221.701323] 990000 / 1171073 nodes visited\n",
            "[221.879604] 1000000 / 1171073 nodes visited\n",
            "[222.033289] 1010000 / 1171073 nodes visited\n",
            "[222.208747] 1020000 / 1171073 nodes visited\n",
            "[222.405852] 1030000 / 1171073 nodes visited\n",
            "[222.606320] 1040000 / 1171073 nodes visited\n",
            "[222.875213] 1050000 / 1171073 nodes visited\n",
            "[223.099145] 1060000 / 1171073 nodes visited\n",
            "[223.293220] 1070000 / 1171073 nodes visited\n",
            "[223.495889] 1080000 / 1171073 nodes visited\n",
            "[223.690780] 1090000 / 1171073 nodes visited\n",
            "[223.910252] 1100000 / 1171073 nodes visited\n",
            "[224.207606] 1110000 / 1171073 nodes visited\n",
            "[224.449193] 1120000 / 1171073 nodes visited\n",
            "[224.741529] 1130000 / 1171073 nodes visited\n",
            "[225.033070] 1140000 / 1171073 nodes visited\n",
            "[225.529095] 1150000 / 1171073 nodes visited\n",
            "[226.357839] 1160000 / 1171073 nodes visited\n",
            "[236.338289] 1170000 / 1171073 nodes visited\n",
            "[241.619840] Concatenation...\n",
            "[241.686739] Renumbering nodes\n",
            "[241.686772] Initial node count 1171073\n",
            "[241.712895] Removed 1033752 null nodes\n",
            "[241.712923] Concatenation over!\n",
            "[241.712926] Clipping short tips off graph, drastic\n",
            "[241.943039] Concatenation...\n",
            "[242.013222] Renumbering nodes\n",
            "[242.013266] Initial node count 137321\n",
            "[242.036677] Removed 16027 null nodes\n",
            "[242.036719] Concatenation over!\n",
            "[242.036724] 121294 nodes left\n",
            "[242.045599] Writing into graph file Assem/Graph...\n",
            "[243.102249] Measuring median coverage depth...\n",
            "[243.199647] Median coverage depth = 36.237288\n",
            "[243.199777] Removing contigs with coverage < 18.118644...\n",
            "[243.269482] Concatenation...\n",
            "[243.537724] Renumbering nodes\n",
            "[243.537756] Initial node count 121294\n",
            "[243.539637] Removed 119785 null nodes\n",
            "[243.539659] Concatenation over!\n",
            "[243.539950] Concatenation...\n",
            "[243.540384] Renumbering nodes\n",
            "[243.540398] Initial node count 1509\n",
            "[243.540411] Removed 0 null nodes\n",
            "[243.540416] Concatenation over!\n",
            "[243.540423] Clipping short tips off graph, drastic\n",
            "[243.540534] Concatenation...\n",
            "[243.541824] Renumbering nodes\n",
            "[243.541838] Initial node count 1509\n",
            "[243.541888] Removed 33 null nodes\n",
            "[243.541895] Concatenation over!\n",
            "[243.541901] 1476 nodes left\n",
            "[243.541907] WARNING: NO EXPECTED COVERAGE PROVIDED\n",
            "[243.541912] Velvet will be unable to resolve any repeats\n",
            "[243.541917] See manual for instructions on how to set the expected coverage parameter\n",
            "[243.542636] Concatenation...\n",
            "[243.542956] Renumbering nodes\n",
            "[243.542967] Initial node count 1476\n",
            "[243.542979] Removed 0 null nodes\n",
            "[243.542985] Concatenation over!\n",
            "[243.542991] Removing reference contigs with coverage < 18.118644...\n",
            "[243.543113] Concatenation...\n",
            "[243.543258] Renumbering nodes\n",
            "[243.543270] Initial node count 1476\n",
            "[243.543282] Removed 0 null nodes\n",
            "[243.543287] Concatenation over!\n",
            "[243.548855] Writing contigs into Assem/contigs.fa...\n",
            "[243.871207] Writing into stats file Assem/stats.txt...\n",
            "[243.877200] Writing into graph file Assem/LastGraph...\n",
            "[244.353147] Estimated Coverage cutoff = 18.118644\n",
            "Final graph has 1476 nodes and n50 of 19714, max 71215, total 4544326, using 0/831196 reads\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Hybrid assembler"
      ],
      "metadata": {
        "id": "_njJBlkkfv0n"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#WenganM 14Min\n",
        "! mkdir WenganM \n",
        "! cd WenganM && perl /content/wengan-v0.2-bin-Linux/wengan.pl -x ontraw -a M -s /content/wengan_demo/ecoli/reads/EC.50X.R1.fastq.gz,/content/wengan_demo/ecoli/reads/EC.50X.R2.fastq.gz -l /content/wengan_demo/ecoli/reads/EC.ONT.30X.fa.gz -p ec_Wm_or1 -t 2 -g 5"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NicaqoXLf0uN",
        "outputId": "b851d38c-871f-4418-9a7d-4cf0489d448d"
      },
      "execution_count": 34,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/wengan-v0.2-bin-Linux/bin/minia -in ec_Wm_or1.minia_reads.41.txt -kmer-size 41 -abundance-min 2 -out ec_Wm_or1.minia.41 -minimizer-size 10 -max-memory 5000 -nb-cores 2 2> ec_Wm_or1.minia.41.err > ec_Wm_or1.minia.41.log\n",
            "rm -f ec_Wm_or1.minia.41.unitigs.fa.glue* ec_Wm_or1.minia.41.h5 ec_Wm_or1.minia.41.unitigs.fa\n",
            "/content/wengan-v0.2-bin-Linux/bin/minia -in ec_Wm_or1.minia_reads.81.txt -kmer-size 81 -abundance-min 2 -out ec_Wm_or1.minia.81 -minimizer-size 10 -max-memory 5000 -nb-cores 2 2> ec_Wm_or1.minia.81.err > ec_Wm_or1.minia.81.log\n",
            "rm -f ec_Wm_or1.minia.81.unitigs.fa.glue* ec_Wm_or1.minia.81.h5 ec_Wm_or1.minia.81.unitigs.fa\n",
            "/content/wengan-v0.2-bin-Linux/bin/minia -in ec_Wm_or1.minia_reads.121.txt -kmer-size 121 -abundance-min 2 -out ec_Wm_or1.minia.121 -minimizer-size 10 -max-memory 5000 -nb-cores 2 2> ec_Wm_or1.minia.121.err > ec_Wm_or1.minia.121.log\n",
            "rm -f ec_Wm_or1.minia.121.unitigs.fa.glue* ec_Wm_or1.minia.121.h5 ec_Wm_or1.minia.121.unitigs.fa\n",
            "/content/wengan-v0.2-bin-Linux/bin/seqtk seq -L 200 ec_Wm_or1.minia.121.contigs.fa  | /content/wengan-v0.2-bin-Linux/bin/seqtk seq -l 60 - > ec_Wm_or1.minia.contigs.fa\n",
            "grep \">\" ec_Wm_or1.minia.contigs.fa | sed 's/km:f://' | awk '{print $1\" \"$4}' | sed 's/>//g' > ec_Wm_or1.minia.contigs.cov.txt\n",
            "/content/wengan-v0.2-bin-Linux/bin/fastmin-sg shortr -c 30 -k 21 -w 10 -q 20 -r 50000 -t 2 ec_Wm_or1.minia.contigs.fa ec_Wm_or1.fms.txt 2>ec_Wm_or1.fms.err >ec_Wm_or1.fms.log\n",
            "/content/wengan-v0.2-bin-Linux/bin/intervalmiss -d 7 --fst 0.1 -b ec_Wm_or1.minia.contigs.cov.txt -t 2 -s ec_Wm_or1.fms.sams.txt -c  ec_Wm_or1.minia.contigs.fa -p  ec_Wm_or1 2>ec_Wm_or1.im.err >ec_Wm_or1.im.log\n",
            "grep \">\" ec_Wm_or1.MBC7.msplit.fa | sed 's/>//' | awk '{print $1\" \"$2}' | sed 's/>//g' > ec_Wm_or1.MBC7.msplit.cov.txt\n",
            "/content/wengan-v0.2-bin-Linux/bin/fastmin-sg ontraw -k 15 -w 5 -q 40 -m 150 -r 300 -t  2 -p ec_Wm_or1 -I  500,1000,2000,3000,4000,5000,6000,7000,8000,10000,15000,20000,30000,40000,50000 ec_Wm_or1.MBC7.msplit.fa ec_Wm_or1.fml.txt 2>ec_Wm_or1.fml.err >ec_Wm_or1.fml.log\n",
            "/content/wengan-v0.2-bin-Linux/bin/liger  --mlp 10000  -t 2 -c  ec_Wm_or1.MBC7.msplit.fa -l  longreads.ec_Wm_or11.fa -d ec_Wm_or1.MBC7.msplit.cov.txt -p ec_Wm_or1 -s ec_Wm_or1.sams.txt 2>ec_Wm_or1.liger.err >ec_Wm_or1.liger.log\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Quast benchmark\n",
        "\n"
      ],
      "metadata": {
        "id": "vlBXu9xmkMBI"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!conda install -y quast"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "77sGgUSwmIYz",
        "outputId": "b2f9aed6-ef70-4de9-e6be-49d2989bfaa5"
      },
      "execution_count": 43,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/bin/conda\", line 7, in <module>\n",
            "    from conda.cli import main\n",
            "ModuleNotFoundError: No module named 'conda'\n"
          ]
        }
      ]
    }
  ]
}